client/components/mma/shared/asyncComponents/DefaultApiResponseHandler.tsx

```diff
@@ -0,0 +1,48 @@
+import type ResponseProcessor from './ResponseProcessor';
+
+export const JsonResponseHandler: ResponseProcessor = (
+	response: Response | Response[],
+) => {
+	return handleResponses(response, (r: Response) => r.json());
+};
+
+export const TextResponseHandler: ResponseProcessor = (
+	response: Response | Response[],
+) => {
+	return handleResponses(response, (r: Response) => r.text());
+};
+
+function handleResponses(
+	response: Response | Response[],
```

Comment:
Is there any better way to handle this so we don't have to cater to a response or an array of responses? It seems quite confusing, for instance the function name is `handleResponses` but the parameter is `response` . Could the calling code handle mapping an array onto individual calls to this function where needed?

Reply:
We're catering for a response or an array of responses to be able to swap out existing uses of asyncLoader which can use either. I think this is the right place to do map the array. Any thoughts @KentGrm? Would it make more sense if the parameter was named `responses`?

Reply:
Ok, it's not a big deal, just prefer to avoid having to check `isArray` throughout the code but it's only in two places I guess.